### ISSUE TITLE
HTTP 400 error when using `{vsn, git}` in src/appname.app.src

### ISSUE THREAD

**Issue by davisp** [inaka/hexer#31]
It looks like hexer doesn't try and auto-resolve the version for the published package when reading the .app.src file.

```
Before publishing, please read Hex CoC: https://hex.pm/docs/codeofconduct
Proceed? [Y/n]: y
Error: {400,
        #{<<"message">> =>     <<"/tmp/build_f3222941c2fa0a473db5c3ef592b3f78/lib/hex_web/release.ex:184: value `\"git\"` in `where` cannot be cast to type HexWeb.Version in query:\n\nfrom r in HexWeb.Release,\n  where: r.package_id in ^[1406],\n  where: r.version == ^\"git\",\n  limit: 1\n\nError when casting value to `HexWeb.Release.version`">>,
          <<"status">> => 400}}
```

Either adding the rebar-like version info or having a command line override option would be acceptable I'd think.

**Comment by elbrujohalcon** [comment]
@davisp good catch! I'll check how erlang.mk resolves such a thing and try to come up with a similar solution… stay tuned ;)

**Comment by davisp** [comment]
One thing I noticed was that hexer likes to include compiled code for NIFs by default. Perhaps if those include globs were tightened up a bit we could just rely on the app's build system to produce `ebin/appname.app` and read that.

Also, along those lines, something that probably deserves its own issue would be something like `hexer test` or `hexer test -- make check` that would create the package and then expand it and run make check in a different directory to ensure that the package builds properly with what's included.

**Comment by elbrujohalcon** [comment]
Closed by #34
